.github/workflows/pr_basic_checks.yml

```diff
@@ -0,0 +1,244 @@
+          git diff origin/${GITHUB_BASE_REF} -U0 | ( grep -a '^+' || true ) | ( ! grep -axv '.*' )
+
+      - 
+        name: "include check"
+        run: |
+          ! git grep '^#include\s["'"']mbed.h['"'"]$' -- '*.c' '*.h' '*.cpp' '*.hpp' \
+              ':!*platform_mbed.h' ':!*TESTS/*' ':!TEST_APPS/' ':!UNITTESTS/' \
+              ':!*tests/*' ':!*targets/*' ':!*TARGET_*' ':!*unsupported/*' \
+              ':!*events/tests/*' ':!*drivers/tests/*'
+
+      - 
+        name: astyle checks
+        run: |
+          git diff --name-only --diff-filter=d origin/${GITHUB_BASE_REF} \
+            | ( grep '.*\.\(c\|cpp\|h\|hpp\)$' || true ) \
+            | ( grep -v -f .codecheckignore || true ) \
+            | while read file; do astyle -n --options=.astylerc "${file}"; done
+          git diff --exit-code --diff-filter=d --color
+
+      - 
+        name: spell checks
+        run: |
+          ./tools/test/travis-ci/doxy-spellchecker/spell.sh drivers .codecheckignore
+          ./tools/test/travis-ci/doxy-spellchecker/spell.sh platform .codecheckignore
+          ./tools/test/travis-ci/doxy-spellchecker/spell.sh events .codecheckignore
+          ./tools/test/travis-ci/doxy-spellchecker/spell.sh rtos .codecheckignore
+          ./tools/test/travis-ci/doxy-spellchecker/spell.sh connectivity/netsocket .codecheckignore
+
+      - 
+        name: doxygen
+        run: |
+          ccache -s
+          mkdir BUILD
+          # Assert that the Doxygen build produced no warnings.
+          # The strange command below asserts that the Doxygen command had an
+          # output of zero length
+          doxygen doxyfile_options 2>&1
+          # Once Mbed OS has been fixed, enable the full test by replacing the top line with this:
+          # - ( ! doxygen doxyfile_options 2>&1 | grep . )
+          # Assert that all binary libraries are named correctly
+          # The strange command below asserts that there are exactly 0 libraries
+          # that do not start with lib
+          find "(" -name "*.a" -or -name "*.ar" ")" -and -not -name "lib*" |
+          tee BUILD/badlibs |
+          sed -e "s/^/Bad library name found: /" && [ ! -s BUILD/badlibs ]
+          # Assert that all assembler files are named correctly
+          # The strange command below asserts that there are exactly 0 libraries
+          # that do end with .s
+          find  -name "*.s" | tee BUILD/badasm |
+          sed -e "s/^/Bad Assembler file name found: /" && [ ! -s BUILD/badasm ]
+
+  python-tests:
+    # these tests run in 3.7, hence running in vm not in pre-built docker
+    runs-on: ubuntu-latest
+
+    steps:
+      -
+        name: Checkout repo
+        uses: actions/checkout@v2
+
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.7'
+
+      -
+        name: install dependencies
+        run: |
+          pip install -r requirements.txt 
+          pip install icetea pytz
```

Comment:
why do we still need to install icetea  stuff

Reply:
we don't need this apparently. I copied this from travis script. Will remove it.